src/subcmd/calculate_with_dune.rs

```diff
@@ -0,0 +1,414 @@
+                                        Ok(j) => j,
+                                        Err(_) => {
+                                            sp.stop_with_message(
+                                                "Error: Failed to parse results".red().to_string(),
+                                            );
+                                            return;
+                                        }
+                                    };
+
+                                    // Extract block rewards from result
+                                    if let Some(rows) = json["result"]["rows"].as_array() {
+                                        for row in rows {
+                                            if let (Some(row_epoch), Some(rewards)) = (
+                                                row["epoch"].as_u64(),
+                                                row["block_rewards"].as_u64(),
+                                            ) {
+                                                if row_epoch == epoch {
+                                                    total_block_rewards = Some(rewards);
+                                                    break;
+                                                }
+                                            }
+                                        }
+                                    }
+
+                                    if total_block_rewards.is_some() {
+                                        break;
+                                    }
+
+                                    sp.stop_with_message(
+                                        format!("Error: No rewards data found for epoch {}", epoch)
+                                            .red()
+                                            .to_string(),
+                                    );
+                                    return;
+                                }
+                                Err(_) => {
+                                    sp.stop_with_message(
+                                        "Error: Failed to fetch results".red().to_string(),
+                                    );
+                                    return;
+                                }
+                            }
+                        }
+                        _ => {
+                            tokio::time::sleep(Duration::from_secs(poll_interval_secs)).await;
+                            continue;
+                        }
+                    }
+                }
+                Err(_) => {
+                    tokio::time::sleep(Duration::from_secs(poll_interval_secs)).await;
+                    continue;
+                }
+            }
+        }
+
+        let total_block_rewards = match total_block_rewards {
+            Some(rewards) => rewards,
+            None => {
+                sp.stop_with_message("Error: Query timed out".red().to_string());
+                return;
+            }
+        };
```

Comment:
have you seen the `duners` crate that abstracts over the dune API and polling? Example usage [here](https://github.com/igneous-labs/manager-fee-trf-crank/blob/master/cli/src/subroutines/dune/update_stake_pool_balance_finder.rs)

Reply:
Although its fine to just do it with raw http requests too. Tho you mightve missed a [`QUERY_STATE_CANCELLED`](https://docs.rs/duners/latest/src/duners/response.rs.html#32-38), which is also terminal state. Not the end of the world since the loop will eventually terminate with timeout anyway but i guess the user would make more dune API requests than needed in this case

Reply:
Ah yeah, I'll add for that and `QUERY_STATE_EXPIRED` as well just to be safe